### PR TITLE
[WIP]add bn with relu fused.

### PR DESCRIPTION
**Pull request by wzzju**
* Add batch normalization with relu fused.

### CODE REVIEW

File: paddle/fluid/operators/batch_norm_op.cu [hunk 1]
```diff
@@ -46,6 +46,7 @@ class BatchNormKernel<platform::CUDADeviceContext, T>
     float momentum = ctx.Attr<float>("momentum");
     const bool is_test = ctx.Attr<bool>("is_test");
     const bool use_global_stats = ctx.Attr<bool>("use_global_stats");
+    const bool fuse_with_relu = ctx.Attr<bool>("fuse_with_relu");
```
Comment (gongweibao) [inline paddle/fluid/operators/batch_norm_op.cu:49]: You can get the supported act_type in `cudnnBatchNormalizationForwardTrainingEx`.
Not only relu.

File: paddle/fluid/operators/batch_norm_op.cu [hunk 2]
```diff
@@ -191,14 +192,31 @@ class BatchNormKernel<platform::CUDADeviceContext, T>
       if ((N * H * W * D) == 1) {
         // Only 1 element in normalization dimension,
         // skip the batch norm calculation, let y = x.
-        framework::TensorCopy(*x, ctx.GetPlace(), y);
+        if (fuse_with_relu) {
+          auto x_v = framework::EigenVector<T>::Flatten(*x);
+          auto y_v = framework::EigenVector<T>::Flatten(*y);
+          auto &dev = *dev_ctx.eigen_device();
+          y_v.device(dev) = x_v.cwiseMax(static_cast<T>(0));
```
Comment (gongweibao) [inline paddle/fluid/operators/batch_norm_op.cu:199]: Maybe use the op's implementation?

**Comment by gongweibao** [comment]
Need resnet50 top1 acc and top5 acc.
